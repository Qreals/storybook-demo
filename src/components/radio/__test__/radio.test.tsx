import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Radio from '../index';
import mountTest from '../../../utils/mount-test';

describe('Radio', () => {
	mountTest('Radio mount unmount', Radio);

	test('it should generate a snapshot', () => {
		const { container } = render(<Radio />);
		expect(container).toMatchSnapshot();
	});

	test('it should tigger change events when value changes', () => {
		const onChange = jest.fn();
		render(<Radio onChange={onChange} />);
		const element = screen.getByRole('radio');
		fireEvent.click(element);
		expect(onChange).toHaveBeenCalledTimes(1);
	});
});
