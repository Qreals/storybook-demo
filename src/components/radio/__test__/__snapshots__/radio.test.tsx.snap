// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio it should generate a snapshot 1`] = `
<div>
  <label
    class="ant-radio-wrapper"
  >
    <span
      class="ant-radio"
    >
      <input
        class="ant-radio-input"
        type="radio"
        value=""
      />
      <span
        class="ant-radio-inner"
      />
    </span>
  </label>
</div>
`;
